### ISSUE TITLE
Unhandled Promise rejection: TypeError: Cannot set property 'name' of undefined

### ISSUE THREAD

**Issue by filipesilva** [angular/zone.js#554]
Using `zone.js@0.7.3`, Angular 2 `routerLink` does not work properly whereas with  `zone.js@0.7.2` it did.

Here is a repro plunker: https://plnkr.co/edit/SYPR2SRi0xayQ693jJAB?p=preview
It is a simple list with a couple of links. Clicking on any link breaks the app with the following console error:
```
zone.js@0.7.3?main=browser:405 Unhandled Promise rejection: TypeError: Cannot set property 'name' of undefined
    at new EmptyError (VM20135 EmptyError.js:21)
    at FirstSubscriber._complete (VM20134 first.js:147)
// ...
```
In `index.html` if you revert `https://unpkg.com/zone.js@0.7.3` to `0.7.2` the app works again.

**Comment by JiaLiPassion** [comment]
@filipesilva , sorry, it is my mistake, I didn't return this in PR #547, I'll make a PR now.

**Comment by filipesilva** [comment]
@JiaLiPassion if I understand your PR correctly, it should only affect errors themselves. This scenario should not result in an error at all though, since it worked with `0.7.2` without throwing errors.

**Comment by chuckjaz** [comment]
This has been fixed with 0.7.4

**Comment by blueskyfish** [comment]
Hi Guys,

Thanks for the solve of the zone.js in version 0.7.4. 👍 :-)
